rmw_fastrtps_cpp/src/rmw_publish.cpp

```diff
@@ -45,17 +46,10 @@ rmw_publish(const rmw_publisher_t * publisher, const void * ros_message)
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(
     info, "publisher info pointer is null", return RMW_RET_ERROR, error_allocator);
 
-  eprosima::fastcdr::FastBuffer buffer;
-  eprosima::fastcdr::Cdr ser(buffer, eprosima::fastcdr::Cdr::DEFAULT_ENDIAN,
-    eprosima::fastcdr::Cdr::DDS_CDR);
-
-  if (!_serialize_ros_message(ros_message, buffer, ser, info->type_support_,
-    info->typesupport_identifier_))
-  {
-    RMW_SET_ERROR_MSG("cannot serialize data");
-    return RMW_RET_ERROR;
-  }
-  if (!info->publisher_->write(&ser)) {
+  rmw_fastrtps_cpp::SerializedData data;
+  data.is_cdr_buffer = false;
+  data.data = const_cast<void *>(ros_message);
+  if (!info->publisher_->write(&data)) {
```

Comment:
shouldn't this be `data.data` ?

Reply:
As stated in [this comment](https://github.com/eProsima/rmw_fastrtps/blob/d4f1dd8453154f02f4fa5ddca798dc9b8bd8bf4a/rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport.hpp#L45), it is not. The write method is provided with a pointer to a SerializedData struct, that will in turn be passed on to the serialize method on TypeSupport_impl.